packages/client/tests/functional/chunking-query/tests.ts

```diff
@@ -0,0 +1,263 @@
+      })
+
+      test('should succeed when the explicit number of params is 32767 or less', async () => {
+        const n = 32767
+        const ids = await createTags(n)
+        const tags = await getTagsParams(ids)
+
+        expect(tags.length).toBe(n)
+      })
+    })
+
+    describeIf(providerFlavor === undefined)('With Rust drivers only', () => {
+      // See: https://github.com/prisma/prisma/issues/21802.
+      test('Selecting 32767 ids at once in two inclusive disjunct filters results in error: "too many bind variables"', async () => {
+        const ids = generatedIds(32767)
+
+        try {
+          await selectWith2InFilters(ids)
+          // unreachable
+          expect(true).toBe(false)
+        } catch (error) {
+          const e = error as Error
+
+          if (['postgresql', 'cockroachdb'].includes(provider)) {
+            expect(e.message).toContain('Assertion violation on the database')
+            expect(e.message).toContain('too many bind variables in prepared statement')
+            expect(e.message).toContain(`expected maximum of 32767, received 65534`)
+          } else {
+            expect(e.message).toContain('Prepared statement contains too many placeholders')
+          }
+        }
+      })
+
+      test('Selecting 32768 ids at once in two inclusive disjunct filters results in error: "too many bind variables"', async () => {
+        const ids = generatedIds(32768)
+
+        try {
+          await selectWith2InFilters(ids)
+          // unreachable
+          expect(true).toBe(false)
+        } catch (error) {
+          const e = error as Error
+
+          if (['postgresql', 'cockroachdb'].includes(provider)) {
+            expect(e.message).toContain('Assertion violation on the database')
+            expect(e.message).toContain('too many bind variables in prepared statement')
+            expect(e.message).toContain(`expected maximum of 32767, received 65535`)
+          } else {
+            expect(e.message).toContain('Prepared statement contains too many placeholders')
+          }
+        }
+      })
+    })
+
+    describeIf(providerFlavor !== undefined)('With Driver Adapters only', () => {
+      test('Selecting 32768 ids at once in two inclusive disjunct filters works', async () => {
+        const ids = generatedIds(32768)
+        await selectWith2InFilters(ids)
+      })
+
+      // See: https://github.com/prisma/prisma/issues/21803.
+      test('Selecting 65536 ids at once in two inclusive disjunct filters results in error', async () => {
+        const ids = generatedIds(65536)
+
+        try {
+          await selectWith2InFilters(ids)
+          // unreachable
+          expect(true).toBe(false)
+        } catch (error) {
+          const e = error as Error
+
+          if (['postgresql', 'cockroachdb'].includes(provider)) {
+            expect(e.message).toContain('bind message has 32767 parameter formats but 0 parameters')
+          }
+        }
+      })
```

Comment:
These are the failing cases, can we wrap them into their own `describe`?
(That would also allow us to move `selectWith2InFilters` into here as it is used in both `describeIf` that will then be children of it)

Reply:
> These are the failing cases

Not all of them are failing: "With Driver Adapters only > Selecting 32768 ids at once in two inclusive disjunct filters works" is succeeding. Where should that be? How many `describe` blocks would you recommend having here?

Reply:
Enough to structure the code in a way that one can understand what the "helper functions" are used for.

Generally these are the queries where chunking currently does not work though, and for driver adapters it only works as the driver can handle more than what our chunking would create (as our chunking is based on the tokio-postgres capabilities, which are artificially low). So a grouping describe "chunking logic does not trigger" would work for me here.

Reply:
(I have implemented this refactoring)